db.go

```diff
@@ -349,22 +377,31 @@ type DB struct {
 	IgnoreUnmappedCols bool
 	Context            context.Context
 	Logger             QueryLogger
-	mu                 sync.RWMutex
-	models             map[reflect.Type]*Model
-	mappings           map[reflect.Type]fieldMap
+	// Query deadline, ignored if Deadline.IsZero().
+	Deadline time.Time
+	// Function to add a deadline to a query.
+	DeadlineQueryRewriter func(db *DB, query interface{}, millis int64) (queryWithDeadline interface{}, err error)
```

Comment:
I'm adding two standard query rewriters that can be installed when the DB is instantiated using functions `db.WithPerconaQueryDeadlines()` or `db.WithMySql57QueryDeadlines()`.